client-base/src/main/kotlin/app/cash/backfila/client/internal/BackfilaStartupConfigurator.kt

```diff
@@ -34,38 +34,49 @@ class BackfilaStartupConfigurator @Inject constructor(
     }
 
     // Send the Backfila registration request
-    val request = ConfigureServiceRequest.Builder()
-      .backfills(
-        registrations.map { registration ->
-          @Suppress("UNCHECKED_CAST")
-          val parameters = backfilaParametersFromClass(registration.parametersClass)
-          ConfigureServiceRequest.BackfillData.Builder()
-            .name(registration.name)
-            .description(registration.description)
-            .parameters(parameters)
-            .delete_by(registration.deleteBy?.toEpochMilli())
-            .build()
-        },
-      )
-      .connector_type(config.connector_type)
-      .connector_extra_data(config.connector_extra_data)
-      .slack_channel(config.slack_channel)
-      .variant(config.variant)
-      .build()
-
-    // TODO(mikepaw): make this async.
-    try {
-      backfilaClient.configureService(request)
-
-      logger.info {
-        "Backfila lifecycle listener initialized. " +
-          "Updated backfila with ${registrations.size} backfills."
-      }
+    val request: ConfigureServiceRequest? = try {
```

Comment:
nit: this try/catch block might be a little easier to read if it was extracted to its own method.

Reply:
Maybe something like:

```suggestion
    val request: ConfigureServiceRequest? = if (backfilaClient.throwOnStartup) {
      getRequest().getOrThrow()
    } else {
      getRequest().getOrNull()
    }
    if (request == null) {
       logger.error(e) { "Exception generating backfila registration request, skipped!" }
    }
```

where:

```kt
private fun getRequest(): Result<ConfigureServiceRequest> = runCatching {
  ConfigureServiceRequest.Builder()...
}
```

Reply:
Yep updated both of those try/catches. Thanks for the suggestion!